src/CloudNative.CloudEvents.Avro/AvroEventFormatter.cs

```diff
@@ -41,30 +41,34 @@ public class AvroEventFormatter : CloudEventFormatter
         private const string DataName = "data";
 
         private static readonly string CloudEventAvroMediaType = MimeUtilities.MediaType + MediaTypeSuffix;
-        private static readonly RecordSchema avroSchema;
-        private static readonly DefaultReader avroReader;
-        private static readonly DefaultWriter avroWriter;
-        
-        static AvroEventFormatter()
+        private readonly IGenericRecordSerializer serializer;
+
+        /// <summary>
+        /// Creates an AvroEventFormatter using the default serializer.
+        /// </summary>
+        public AvroEventFormatter() : this(new BasicGenericRecordSerializer()) { }
+
+        /// <summary>
+        /// Creates an AvroEventFormatter that uses a custom <see cref="IGenericRecordSerializer"/>.
+        /// </summary>
+        /// <remarks>
+        /// It is recommended to use the default serializer before defining your own wherever possible.
+        /// </remarks>
+        public AvroEventFormatter(IGenericRecordSerializer genericRecordSerializer)
         {
-            // We're going to confidently assume that the embedded schema works. If not, type initialization
-            // will fail and that's okay since the type is useless without the proper schema.
-            using (var sr = new StreamReader(typeof(AvroEventFormatter).Assembly.GetManifestResourceStream("CloudNative.CloudEvents.Avro.AvroSchema.json")))
-            {
-                avroSchema = (RecordSchema) Schema.Parse(sr.ReadToEnd());
-            }
-            avroReader = new DefaultReader(avroSchema, avroSchema);
-            avroWriter = new DefaultWriter(avroSchema);
+            serializer = genericRecordSerializer;
         }
 
+        /// <summary>
+        /// Avro schema used to serialize and deserialize the CloudEvent.
+        /// </summary>
+        public static readonly RecordSchema AvroSchema = ParseEmbeddedSchema();
```

Comment:
Eek, so sorry not to have noticed this before - this is currently a field. Please make it a property, just by adding `{ get; }` before the `=`. Glad I spotted it now. (Changing it later would be a breaking change.)

Reply:
No worries, I've amended my commit to add that change. It's my bad, I initially thought a field made more sense here.